### PR TITLE
Added API test cases.

### PR DESCRIPTION
**Pull request by Pallavisawale123**
Added Demo API test cases using java ,spring boot ,Rest Assured, API report generation using extent library

### CODE REVIEW

File: reports/Sun Dec 17 17-42-33 IST 2023/API_Execution_Automation.html
Comment (vikassanap) [inline reports/Sun Dec 17 17-42-33 IST 2023/API_Execution_Automation.html:1]: exclude this file and folder using .gitignore file
You should never push temp files

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java [hunk 1]
```diff
@@ -63,7 +58,7 @@ public void validatePostWithString() {
                 .body("{\"name\":\"pallavi\",\"job\":\"QA\"}")
```
Comment (vikassanap) [inline src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: use JsonObect to construct this kind of strings

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java [hunk 2]
```diff
@@ -82,7 +77,7 @@ public void validatePutWithString() {
                 .body("{\"name\":\"pallavisawale\",\"job\":\"SDET\"}")
```
Comment (vikassanap) [inline src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: this is very primitive way to execute APIs, its always better to use requestSpecBuilder and responseSpecBuilder in restassured.

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java [hunk 3]
```diff
@@ -63,7 +58,7 @@ public void validatePostWithString() {
                 .body("{\"name\":\"pallavi\",\"job\":\"QA\"}")
                 .when()
                 .post("https://reqres.in/api/users");
-        AssertHelper.assertEquals(response.getStatusCode(), StatusCode.CREATED.code,"");
+        AssertHelper.assertEquals(response.getStatusCode(), StatusCode.CREATED.code, "");
```
Comment (vikassanap) [inline src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: you should always add a message

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java [hunk 4]
```diff
@@ -63,7 +58,7 @@ public void validatePostWithString() {
                 .body("{\"name\":\"pallavi\",\"job\":\"QA\"}")
                 .when()
                 .post("https://reqres.in/api/users");
```
Comment (vikassanap) [inline src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: path should be kept in properties

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java [hunk 5]
```diff
@@ -82,7 +77,7 @@ public void validatePutWithString() {
                 .body("{\"name\":\"pallavisawale\",\"job\":\"SDET\"}")
                 .when()
                 .put("https://reqres.in/api/users/2");
-        AssertHelper.assertEquals(response.getStatusCode(), StatusCode.SUCCESS.code,"");
+        AssertHelper.assertEquals(response.getStatusCode(), StatusCode.SUCCESS.code, "");
```
Comment (vikassanap) [inline src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java:154]: assertions should contains:
1. Status code
2. Response body
3. Response header if required

**Review by vikassanap — CHANGES_REQUESTED** [review]

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java (hunk 1 above)
Comment (Pallavisawale123) [reply src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: Updated changes .

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java (hunk 2 above)
Comment (Pallavisawale123) [reply src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: Updated test cases..

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java (hunk 3 above)
Comment (Pallavisawale123) [reply src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: Added verification message..

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java (hunk 4 above)
Comment (Pallavisawale123) [reply src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java]: Uses the path from properties file.

File: src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java (hunk 5 above)
Comment (Pallavisawale123) [reply src/test/java/com/Automation/APIAutomation/UserManagement/postUsers.java:154]: Added validation using response

File: src/test/resources/application.properties
Comment (vikassanap) [inline src/test/resources/application.properties:1]: you can still use this file to maintain baseURL, api paths, etc

Comment (Pallavisawale123) [reply src/test/resources/application.properties:1]: Added src/test/resources/application.properties ,I'll use this folder for further test cases.

**Pull request merged**